MatrixSDK/MXRestClient.m

```diff
@@ -2416,14 +2419,21 @@ - (MXHTTPOperation*)joinRoom:(NSString*)roomIdOrAlias
         for (NSString *viaServer in viaServers)
         {
             NSString *value = [MXTools encodeURIComponent:viaServer];
+            NSString *parameter = [NSString stringWithFormat:@"server_name=%@", value];
```

Comment:
Optional improvement, but it would be nice to extract this now into `MXTools` or somewhere else where the code can be tested. The code is not super complicated, but does include some magical constants (2048, +1) and it would be good to validate with a test.